CONTRIBUTING.md

```diff
@@ -33,109 +41,210 @@ For an introduction to the basics of reST syntax, the source format in which Spy
+
+To build just the docs for the current version, run the following on a system with ``make`` (macOS/Linux):
 
 ```bash
 make docs
 make serve
 ```
 
-or, if on Windows:
+To build the full site with the documentation for all Spyder versions, run:
 
-```cmd
-cd doc
-make.bat html
-cd ..
+```bash
+make multidocs
 ```
 
-**NOTE:** You can try to use the `make` commands on Windows if you have `make` installed and something like `Cygwin`, `MSYS2` or `MYSYS`, or by using Windows Subsystem for Linux (`WSL`).
+On a system without ``make`` (like Windows, by default), or to build the docs manually, run:
 
-To build the full site with the documentation for all Spyder versions, run:
+```shell
+python -m sphinx -n -W --keep-going -b html doc doc/_build/html
+```
 
-```bash
-make multidocs
+If not opened automatically with ``make serve``, you should be able to view the HTML output inside the resulting ``_build/html`` directory the build commands produce; ``index.html`` is the main page.
+
+
+
+## Deciding Which Branch to Use
+
+When you start to work on a new pull request (PR), you need to be sure that your work is done on top of the correct branch, and that you base your PR on Github against it.
+
+To guide you, issues on Github are marked with a milestone that indicates the correct branch to use.
+If not, follow these guidelines:
+
+* The ``master`` branch contains the in-development documentation for Spyder 5; most general changes and those specific to that version should be made against this branch.
+* The ``4.x`` branch contains the docs for Spyder 4 and is in bugfix-only mode; no substantial new content should be added here at this point.
+* The ``3.x`` branch is frozen, containing the docs for the legacy Spyder 3 version; no further PRs will be accepted
+
+Of course, if an issue is only present in a specific branch, please base your PR on that branch.
+If you are at all unsure which branch to use, we'll be happy to guide you.
+
+
+
+## Making Your Changes
+
+To start working on a new PR, you need to execute these commands, filling in the branch names where appropriate (``<BASE-BRANCH>`` is the branch you're basing your work against, e.g. ``master``, while ``<FEATURE-BRANCH>`` is the branch you'll be creating to store your changes, e.g. ``fix-doc-typo`` or ``add-plugin-guide``:
+
+```shell
+git checkout <BASE-BRANCH>
+git pull upstream <BASE-BRANCH>
+git checkout -b <FEATURE-BRANCH>
 ```
 
-Either way, you should be able to view the HTML output inside the resulting ``_build`` directory the build commands produce; ``index.html`` is the main page.
+Once you've made and tested your changes, commit them with a descriptive, unique message of 74 characters or less written in the imperative tense, with a capitalized first letter and no period at the end.
+Try to make your commit message understandable on its own, giving the reader a high-level idea of what your changes accomplished without having to dig into the diffs.
```

Comment:
```suggestion
Try to make your commit message understandable on its own, giving the reader a high-level idea of what your changes accomplished without having to dig into the diff of your commit.
```

To be more explicit.

Reply:
```suggestion
Try to make your commit message understandable on its own, giving the reader a high-level idea of what your changes accomplished without having to dig into the full diff output.
```

Avoid repetition while improving clarity